inc/config.class.php

```diff
@@ -226,6 +226,10 @@ function prepareInputForUpdate($input) {
          }
       }
 
+      if (isset($input['devices_in_menu'])) {
```

Comment:
This is not working if there is no device selected, as field will not be set.
My proposal:
```diff
diff --git a/inc/config.class.php b/inc/config.class.php
index ca6c3a0e16..94db69223a 100644
--- a/inc/config.class.php
+++ b/inc/config.class.php
@@ -226,8 +226,10 @@ class Config extends CommonDBTM {
          }
       }
 
-      if (isset($input['devices_in_menu'])) {
-         $input['devices_in_menu'] = exportArrayToDB($input['devices_in_menu']);
+      if (isset($input['_update_devices_in_menu'])) {
+         $input['devices_in_menu'] = exportArrayToDB(
+            isset($input['devices_in_menu']) ? $input['devices_in_menu'] : []
+         );
       }
 
       // lock mechanism update
@@ -539,7 +541,7 @@ class Config extends CommonDBTM {
 
       echo "<tr class='tab_bg_2'>";
       echo "<td><label for='devices_in_menu$rand'>".__('Devices displayed in menu')."</label></td>";
-      echo "<td>";
+      echo "<td><input type='hidden' name='_update_devices_in_menu' value='1'>";
 
       $dd_params = [
          'name'      => 'devices_in_menu',
```